aws/cloudformation/components/ami.yml.erb

```diff
@@ -1,4 +1,40 @@
-<% ami = commit[0..4] -%>
+<%
+# Template component to deploy AMI-backed frontends to an autoscaling group with rolling updates.
+
+# String to append to AMI logical resources that get replaced on every build.
+# Typically a portion of the commit hash.
+ami ||= raise('ami string required')
+
+# Shell script to run when building/updating the AMI-creating instance.
+build_ami ||= <<SH
+INSTANCE_ID=$(curl -s http://169.254.169.254/latest/meta-data/instance-id)
+STACK=${AWS::StackName}
+RESOURCE_ID=AMICreate#{ami}
+STATUS=SUCCESS
+REGION=${AWS::Region}
+SH
```

Comment:
Most of this boilerplate is needed to maintain zero-diff against the build script in existing stacks. We can substantially simplify this in a future PR that will trigger changes to the build script, but I'd like to merge the bulk of these changes first without affecting any existing resources.